src/Entity.php

```diff
@@ -144,14 +153,12 @@ class Entity extends CommonTreeDropdown
             'suppliers_as_private', 'autopurge_delay', 'anonymize_support_agents', 'display_users_initials',
             'contracts_strategy_default', 'contracts_id_default'
         ],
-      // Configuration
+        // Configuration
         'config' => ['enable_custom_css', 'custom_css_code']
     ];
 
-
-    public function getForbiddenStandardMassiveAction()
+    public function getForbiddenStandardMassiveAction(): array
```

Comment:
I would like to always have a typed return value and typed parameters, but for the `CommonGLPI`/`CommonDBTM` methods that are expected to be overriden by many child classes, I think these changes should be done globally, i.e. all classes at once, with a mention in the changelog and maybe also an update in the documentation, the empty plugin, and maybe also a small paragraph in the migration guide.

For instance, when a parent class method signature changes, every child class may have to fix the signature of overriden methods.
```php
class A {
    public function prepareInputForUpdate(): false|array {
        return [];
    }
}
class B extends A {
    public function prepareInputForUpdate() {
        return [];
    }
}
// Fatal error: Declaration of B::prepareInputForUpdate() must be compatible with A::prepareInputForUpdate(): array|false
```

We do not expect any plugin to extends the Entity class, but, IMHO, it will be simplier to add "strict" signatures on `CommonGLPI`/`CommonDBTM` methods and all their child class at once rather than trying to figure, for every class, if a plugin extends it or not. Also, changing the signatures at the abstract classes level is safer as PHPStan will detect every vlass that has not been updated correctly.

Reply:
Fair enough. I figured starting to make the function signatures "correct" in some of the classes that aren't extended (or not often) would not be an issue, but ideally it **would** be done at the CommonDBTM/CommonGLPI level. Not sure if this is the version to do it in or not though. I would certainly be happier to see less yellow (warnings) in the gutter in my IDE for almost every file and work towards a higher PHPStan level.

Reply:
It would be great to do it in GLPI 11.0. PRs will be pretty huge, so each method should probably be migrated in a separated PR, or grouped for methods that are similar (e.g. `prepareInputFor*` could be done in a PR, `post_*Item` in another PR, and so on), to simplify code review.

Could you revert these changes from the current PR ?